src/database-layer/db.ts

```diff
@@ -707,9 +714,55 @@ if (maybePg != null) {
 				`);
 			}
 		}
+
+		let listenerClient: Pg.PoolClient | null = null;
+		async function listen(
+			channel: string,
+			fn: (...args: any[]) => Promise<void>,
+		) {
+			while (listenerClient == null) {
+				try {
+					listenerClient = await pool.connect();
+					listenerClient.on('end', async () => {
+						listenerClient = null;
```

Comment:
I think you need to use `listenerClient.release()` to return it to the pool (probably supplying the error object to have the pool terminate/reconnect it via `listenerClient.release(err)`) otherwise I think there could still be pool exhaustion issues

Reply:
I did that at first, but it would always error saying that the client was already released. I can add it as a "just in case" measure though in a try catch. My understanding is that the client is already released when the end event is emitted, but I don't mind having it there just in case.